draft-ietf-quic-recovery.md

```diff
@@ -506,17 +506,124 @@ Pseudocode for DetectLostPackets follows:
 
      // Inform the congestion controller of lost packets and
      // lets it decide whether to retransmit immediately.
-     OnPacketsLost(lost_packets)
-     foreach (packet in lost_packets)
-       sent_packets.remove(packet.packet_number)
+     if (!lost_packets.empty())
+       OnPacketsLost(lost_packets)
+       foreach (packet in lost_packets)
+         sent_packets.remove(packet.packet_number)
 ~~~
 
 # Congestion Control
 
-(describe NewReno-style congestion control {{?RFC6582}} for QUIC.)
-(describe appropriate byte counting.)
-(define recovery based on packet numbers.)
-(describe min_rtt based hystart.)
+QUIC's congestion control is based on TCP NewReno{{?RFC6582}}
+congestion control to determine the congestion window and pacing rate.
+
+## Slow Start
+
+QUIC uses a slow start approach where the congestion window is increased
+by the same number of bytes as are acknowledged while in slow start.
+QUIC begins every connection in slow start and exits slow start upon
+loss.
+
+## Recovery
+
+Recovery is a period of time beginning with detection of a lost packet.
+It ends when all packets outstanding at the time recovery began have been
+acknowledged or lost. During recovery, the congestion window is not
+increased or decreased.
+
+## Constants of interest
+
+Constants used in congestion control are based on a combination of RFCs,
+papers, and common practice.  Some may need to be changed or negotiated
+in order to better suit a variety of environments.
+
+kInitialWindow (default 10 * 1500 bytes):
+: Default limit on the amount of outstanding data in bytes.
+
+kLossReductionFactor (default 0.5):
+: Reduction in congestion window when a new loss event is detected.
+
+
+## Variables of interest
+
+Variables required to implement the congestion control mechanisms
+are described in this section.
+
+bytes_in_flight:
```

Comment:
What exactly is this measuring: IP, UDP and QUIC header? ACK's, or just the sum of retransmittable frames, or just stream payload?